tinygrad/codegen/linearizer.py

```diff
@@ -86,34 +86,37 @@ def const(self, b:Union[int,float], dtype=dtypes.int32) -> UOp: return self.uop(
     AndNode: lambda self,ops,ctx: functools.reduce(lambda a,b: ctx.uop_alu_idx(a, b, ops, ctx, BinaryOps.MUL, dtype=dtypes.bool), self.nodes[1:], self.nodes[0].render(ops,ctx)) }
 
   def global_load(self, i:int, idxs:Sequence[VariableOrNum], acc=None) -> List[UOp]:
+    for idx in idxs:
+      if any([v.expr is None for v in idx.vars() if not isinstance(idx, Variable)]):
+        print('wtf', idxs)
+        assert False
+
     const = self.bufs[i].realized._buf if isinstance(self.bufs[i].realized, RawConst) else acc
 
     expanded_nodes = [idx.expand() for idx in idxs]
     _idxs = [x[::-1] for x in itertools.product(*expanded_nodes[::-1])]
     upcast_dim = self.get_upcast_dim(i)
 
-    amt = 1
+    amt, dim = 1, None
     if len(upcast_dim) == 1 and len(expanded_nodes[upcast_dim[0]]) in [4,2]:
       dim, amt = upcast_dim[0], len(expanded_nodes[upcast_dim[0]])
 
     # calculate expr_idxs using placeholder variables
-    fake_idxs = [idx if isinstance(idx, NumNode) else Variable(f"_uidx{i}", idx.min, idx.max) for i, idx in enumerate(idxs)]
+    fake_idxs = [(Variable(f"_uidx{j}", idx.min, idx.max) // (1 if j != dim else amt)) if None in [v.expr for v in idx.vars()] else idx for j, idx in enumerate(idxs)]
```

Comment:
Avoids iterating full list if None is found
```suggestion
    fake_idxs = [(Variable(f"_uidx{j}", idx.min, idx.max) // (1 if j != dim else amt)) if any(v.expr is None for v in idx.vars()) else idx for j, idx in enumerate(idxs)]
```

Reply:
Happens more below. Not sure how long idx.vars() is. Memorizing might be nice?